pkg/operators/process/process.go

```diff
@@ -0,0 +1,777 @@
+		for scanner.Scan() {
+			line := scanner.Bytes()
+			idx := bytes.Index(line, []byte(":\t"))
+			if idx == -1 {
+				continue
+			}
+
+			key := string(line[:idx])
+			value := string(line[idx+2:])
+
+			switch key {
+			case "Name":
+				if p.enabledFields[fieldCommand] {
+					procInfo.Command = unescapeCommand(value)
+				}
+			case "PPid":
+				if p.enabledFields[fieldPPID] {
+					ppid64, err := strconv.ParseInt(value, 10, 32)
+					if err == nil {
+						procInfo.PPID = int(ppid64)
+					}
+				}
+			case "State":
+				if p.enabledFields[fieldState] {
+					// First character is the state code
+					if len(value) > 0 {
+						procInfo.State = string(value[0])
+					}
+				}
+			case "Uid":
+				if p.enabledFields[fieldUid] {
+					// First UID is the real UID
+					uidParts := strings.Fields(value)
+					if len(uidParts) > 0 {
+						// Just use the UID as is
+						uid, _ := strconv.ParseUint(uidParts[0], 10, 32)
+						procInfo.Uid = uint32(uid)
+					}
+				}
+			case "VmSize":
+				if p.enabledFields[fieldMemoryVirtual] {
+					// Format: "123456 kB"
+					vmSizeParts := strings.Fields(value)
+					if len(vmSizeParts) > 0 {
+						vmSize, err := strconv.ParseUint(vmSizeParts[0], 10, 64)
+						if err != nil {
+							// Skip this field if parsing fails
+							continue
+						}
+						// Convert from kB to bytes
+						procInfo.MemoryVirtual = vmSize * 1024
+					}
+				}
+			case "VmRSS":
+				if p.enabledFields[fieldMemoryRSS] {
+					// Format: "123456 kB"
+					vmRSSParts := strings.Fields(value)
+					if len(vmRSSParts) > 0 {
+						vmRSS, err := strconv.ParseUint(vmRSSParts[0], 10, 64)
+						if err != nil {
+							// Skip this field if parsing fails
+							continue
+						}
+						// Convert from kB to bytes
+						procInfo.MemoryRSS = vmRSS * 1024
+					}
+				}
+			case "Threads":
+				if p.enabledFields[fieldThreadCount] {
+					threads64, err := strconv.ParseInt(value, 10, 32)
+					if err != nil {
+						// Skip this field if parsing fails
+						continue
+					}
+					procInfo.ThreadCount = int(threads64)
+				}
+			}
+		}
+
+		// Check for scanner errors
+		if err := scanner.Err(); err != nil {
+			return procInfo, fmt.Errorf("scanning status file for PID %d: %w", pid, err)
+		}
+	}
+
+	// Read process stat for CPU usage and start time if needed
+	if p.needStat {
+		statFile := filepath.Join(host.HostProcFs, fmt.Sprintf("%d/stat", pid))
+		statData, err := os.ReadFile(statFile)
+		if err != nil {
+			return procInfo, fmt.Errorf("reading stat file for PID %d: %w", pid, err)
+		}
+		if len(statData) == 0 {
+			return procInfo, fmt.Errorf("empty stat file for PID %d", pid)
+		}
+
+		// Parse stat file - we need to handle the comm field specially
+		// The comm field is enclosed in parentheses and may contain spaces
+		statFields := make([][]byte, 22)
+		err = parseStatFile(statData, statFields)
+		if err != nil {
+			return procInfo, fmt.Errorf("parsing stat file: %w", err)
+		}
+		if len(statFields) >= 22 {
```

Comment:
Why do we need this check? Can't it be less than 22 is we're using 22 on L619?

Reply:
We don't anymore - I changed the logic a bit and this is a leftover.